src/coreclr/src/jit/lsraxarch.cpp

```diff
@@ -2612,9 +2591,12 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
             case NI_AVX2_GatherVector256:
             {
                 assert(numArgs == 3);
+                assert(!isRMW);
+
                 // Any pair of the index, mask, or destination registers should be different
                 srcCount += BuildOperandUses(op1);
                 srcCount += BuildDelayFreeUses(op2);
+                srcCount += BuildDelayFreeUses(op3);
```

Comment:
It should be, but we weren't asserting that or otherwise handling this operand.

This is one of the few HWIntrinsics where the fallback is in managed code: https://source.dot.net/#System.Private.CoreLib/Avx2.cs,0e4c3b2425ab813c and so `op3` is "effectively" guaranteed to be a constant

Reply:
Previously `op3` was being marked as contained, and aw asserting in codegen. Now it appears that we will always generate that constant into a register since it's not contained. Yet the codegen doesn't seem to have changed, so that register will be unused.

Reply:
> Now it appears that we will always generate that constant into a register since it's not contained

I'm not sure why it would no longer be contained? I can change this to be `assert(op3->isContained())` if that makes it clearer.
I had just noticed the handling for `op3` (and `op5` below) wasn't here without a comment or assert covering it.